release-content/0.16/release-notes/17840_Parallel_Transform_Propagation.md

```diff
@@ -1,4 +1,31 @@
-<!-- Parallel Transform Propagation -->
-<!-- https://github.com/bevyengine/bevy/pull/17840 -->
+Transforms in Bevy (and other 3D software) come in two flavors:
 
-<!-- TODO -->
+1. Global transforms ([`GlobalTransform`]), which represent the absolute world-space position, scale and rotation of an object.
+2. Local transforms ([`Transform`]), which represent the position, scale and rotation of an object relative to its parent.
+
+In order to compute the global transform of each object (which is what rendering and physics care about!),
+we need to recursively combine the local transforms of all of our objects down the parent-child hierarchy.
+This process, known as transform propagation, has always been one of the most computationally intensive operations in most Bevy applications.
+Turns out, virtually everything in your world is going to have a transform!
+
+This made transform propagation a prime candidate for some serious optimization work, and Bevy 0.16 comes with *two* impressive performance optimizations.
+
+The first optimization improves our parallelization strategies. While we were already going wide here,
+better work sharing, parallelization across trees and a leaf vs non-leaf split to optimize cache coherency made a huge difference.
+
+The second optimization focuses on saving work for trees where none of the objects have moved.
+In many cases, this is the overwhelming majority of objects: level geometry and props are not typically moving around each frame!
+We're now propagating a "dirty bit" up the hierarchy towards ancestors; allowing transform propagation can ignore entire subtrees of the hierarchy if they encounter an entity without the dirty bit.
+
+The results speak for themselves: taken together, our testing on the incredibly beefy [Caldera Hotel] from Call of Duty: Warzone shows that transform propagation took 1.1 ms in 0.15, and 0.1 ms after these changes in 0.16.
```

Comment:
We should note that Caldera is entirely static.

Tbh we should probably check how much perf impact we had on entirely dynamic scenes...

Reply:
Yeah I can test many_foxes or something.

Reply:
It is significantly faster thanks to parallel transform propagation.

Reply:
About 3-10x faster, depending on the hierarchy.

Reply:
See: https://github.com/bevyengine/bevy/pull/17840

Caldera was 3-7x faster on my machine.